common/src/main/java/org/terraform/main/TerraformGeneratorPlugin.java

```diff
@@ -1,146 +1,147 @@
-                    fixed++;
-                }
-                logger.info("&aSuccessfully finished fixing " + fixed + " pre-mature chunks!");
-
-            }
-        }
-    }
-
-    @EventHandler
-    public void onWorldInit(WorldInitEvent event) {
-        if (event.getWorld().getGenerator() instanceof TerraformGenerator) {
-            logger.info("Detected world: " + event.getWorld().getName() + ", commencing injection... ");
-            if (injector.attemptInject(event.getWorld())) {
-                INJECTED_WORLDS.add(event.getWorld().getName());
-                logger.info("&aInjection success! Proceeding with generation.");
-
-            } else {
-                logger.error("&cInjection failed.");
-            }
-        }
-    }
-
-    @Override
-    public ChunkGenerator getDefaultWorldGenerator(String worldName, String id) {
-        return new TerraformGenerator();
-    }
-
-}
+package org.terraform.main;
+
+import io.papermc.lib.PaperLib;
+import org.bukkit.Bukkit;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.world.WorldInitEvent;
+import org.bukkit.event.world.WorldLoadEvent;
+import org.bukkit.generator.ChunkGenerator;
+import org.drycell.main.DrycellPlugin;
+import org.terraform.coregen.NMSInjectorAbstract;
+import org.terraform.coregen.PopulatorDataPostGen;
+import org.terraform.coregen.TerraformPopulator;
+import org.terraform.coregen.bukkit.NativeGeneratorPatcherPopulator;
+import org.terraform.coregen.bukkit.TerraformGenerator;
+import org.terraform.data.SimpleChunkLocation;
+import org.terraform.data.TerraformWorld;
+import org.terraform.reflection.Post14PrivateFieldHandler;
+import org.terraform.reflection.Pre14PrivateFieldHandler;
+import org.terraform.reflection.PrivateFieldHandler;
+import org.terraform.schematic.SchematicListener;
+import org.terraform.structure.StructureRegistry;
+import org.terraform.tree.SaplingOverrider;
+import org.terraform.utils.version.Version;
+
+import java.lang.reflect.Field;
+import java.util.HashSet;
+import java.util.Random;
+import java.util.Set;
+
+public class TerraformGeneratorPlugin extends DrycellPlugin implements Listener {
+
+    public static final Set<String> INJECTED_WORLDS = new HashSet<>();
+    public static final PrivateFieldHandler privateFieldHandler;
+    public static NMSInjectorAbstract injector;
+    private static TerraformGeneratorPlugin instance;
+
+    static {
+        PrivateFieldHandler handler;
+        try {
+            Field.class.getDeclaredField("modifiers");
+            handler = new Pre14PrivateFieldHandler();
+        } catch (NoSuchFieldException | SecurityException ex) {
+            handler = new Post14PrivateFieldHandler();
+        }
+        privateFieldHandler = handler;
+    }
+
+    public static TerraformGeneratorPlugin get() {
+        return instance;
+    }
+
+    @Override
+    public void onEnable() {
+        super.onEnable();
```

Comment:
What does this super actually do? Is it even nessesary? I never used it in any of my plugins and didn't encounter any issues whatsoever, so unless TfG is doing some stuff that requires this would I say that it isn't needed at all.

Reply:
TFG uses a lib called DryCell, which (i assume) does things in its onEnable method. I just added the paperlib call and didn't touch anything else.

Reply:
Still doesn't explain to me why the usage of `super.onEnable();` is needed here at all.

Reply:
It is needed. Dfsek is right, instead of extending JavaPlugin directly, TerraformGeneratorPlugin extends DrycellPlugin, which does some init stuff in its onEnable. Super.onEnable is called to make sure the same init is done.

Reply:
I see. Didn't saw the `extends DryCellPlugin`.

What are actual benefits compared to stuff like normal (soft)depend and then get the info in the onEnable?

Reply:
Drycell is just a plugin library, so the init stuff is just for simple things like config and command manager initiations etc